### PR TITLE
Adds http.redirect service

### PR DESCRIPTION
**Pull request by DoruM**
This service allows HTTP redirects to HTTPS url when they access a secure service on port 80.  This is a common case for HTTP Proxy configurations which we are now supporting.

### CODE REVIEW

File: server/src/main/xsd/gateway-config-201606.xsd [hunk 1]
```diff
@@ -1168,6 +1168,60 @@
...
+                             name="description"
+                             type="gateway:CollapsedString"/>
+
+                    <element maxOccurs="unbounded" minOccurs="1"
+                             name="accept"
+                             type="anyURI"/>
+
+                    <!-- connect is not supported for redirect -->
+
+                    <element maxOccurs="unbounded" minOccurs="0"
```
Comment (dpwspoon) [inline server/src/main/xsd/gateway-config-201606.xsd]: Is there any real use case for balance??

File: server/src/main/xsd/gateway-config-201606.xsd [hunk 2]
```diff
@@ -1168,6 +1168,60 @@
...
+
+                    <element maxOccurs="1" minOccurs="0"
+                             name="realm-name"
+                             type="gateway:CollapsedString"/>
+
+                    <element maxOccurs="unbounded" minOccurs="0"
+                             name="authorization-constraint"
+                             type="gateway:AuthorizationConstraintType"/>
+
+                    <element maxOccurs="unbounded" minOccurs="0"
```
Comment (dpwspoon) [inline server/src/main/xsd/gateway-config-201606.xsd]: Does cross-site-constraints mean anything for 302

File: service/http.redirect/src/main/java/org/kaazing/gateway/service/http/redirect/HttpRedirectServiceHandler.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        RFC822_FORMAT_PATTERN.setTimeZone(TimeZone.getTimeZone("GMT"));
+    }
+
+    HttpRedirectServiceHandler(Logger logger) {
+        this.logger = logger;
+    }
+
+    @Override
+    public void doSessionCreated(HttpAcceptSession session) throws Exception {
+        // NOOP no license check needed
```
Comment (dpwspoon) [inline service/http.redirect/src/main/java/org/kaazing/gateway/service/http/redirect/HttpRedirectServiceHandler.java]: I don't think  `no license check needed` are needed, is this from copy paste maybe.

Nit: Also, I don't think there is great value in having the `@Override NOOP` method, though I'm ok with them.  I don't want to dig into the API, but perhaps there is a more complete Adapter to use that doesn't require overriding them

File: service/http.redirect/src/main/java/org/kaazing/gateway/service/http/redirect/HttpRedirectServiceHandler.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        session.close(true);
+    }
+
+    @Override
+    protected void doSessionOpened(HttpAcceptSession session) throws Exception {
+        // Get the method used for this request; if not a GET, refuse the
+        // request (KG-1233).
+        // (KG-11211) Enabling HEAD method too.
+        HttpMethod method = session.getMethod();
+        if (method != HttpMethod.GET && method != HttpMethod.HEAD) {
```
Comment (dpwspoon) [inline service/http.redirect/src/main/java/org/kaazing/gateway/service/http/redirect/HttpRedirectServiceHandler.java]: Why just these methods: copy paste from directory service or actual reason.  IIRC correctly, user agents (browsers) might have different behavior on 3xx based on methods, but I don't think that changes what the redirect service should do

**Comment by dpwspoon** [comment]
Don't forget to add the jar to the distribution

File: server/src/main/xsd/gateway-config-201606.xsd (hunk 1 above)
Comment (DoruM) [inline server/src/main/xsd/gateway-config-201606.xsd]: No.

Comment (dpwspoon) [inline server/src/main/xsd/gateway-config-201606.xsd]: let's remove it then

File: service/http.redirect/src/main/java/org/kaazing/gateway/service/http/redirect/HttpRedirectServiceHandler.java (hunk 4 above)
Comment (DoruM) [inline service/http.redirect/src/main/java/org/kaazing/gateway/service/http/redirect/HttpRedirectServiceHandler.java]: Copy paste from directory service. I haven't found anything related to methods in this particular case. I shall remove them.

File: server/src/main/xsd/gateway-config-201606.xsd (hunk 2 above)
Comment (DoruM) [inline server/src/main/xsd/gateway-config-201606.xsd]: Removed.

File: service/http.redirect/src/main/java/org/kaazing/gateway/service/http/redirect/HttpRedirectServiceHandler.java [hunk 5]
```diff
@@ -64,12 +63,6 @@ protected void doSessionOpened(HttpAcceptSession session) throws Exception {
         // Get the method used for this request; if not a GET, refuse the
```
Comment (dpwspoon) [inline service/http.redirect/src/main/java/org/kaazing/gateway/service/http/redirect/HttpRedirectServiceHandler.java]: Nit, please remove comment

**Comment by dpwspoon** [comment]
LGTM, other than comment that should be removed and has no relevance:
```
// Get the method used for this request; if not a GET, refuse the
// request (KG-1233).
// (KG-11211) Enabling HEAD method too.
```

**Comment by sbadugu** [comment]
@DoruM 
The following test is failing on travis:
```
Tests in error: 
  HttpRedirectServiceIT.shouldAcceptSecureLocation Â» Specification Internal erro...

Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

**Comment by DoruM** [comment]
Yes, because K3PO was not merged. I shall remove it from K3PO scripts and add it to this module.

**Pull request merged**